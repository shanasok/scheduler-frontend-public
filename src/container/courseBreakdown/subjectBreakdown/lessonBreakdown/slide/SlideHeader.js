import React from 'react';
import TeacherEditSlideButton from "../../../../teacher/TeacherEditSlideButton";
import TeacherCancelEditSlideButton from "../../../../teacher/TeacherCancelEditSlideButton";
import TeacherSaveEditsSlideButton from "../../../../teacher/TeacherSaveEditsSlideButton";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import {TeacherAddContentButton} from "../../../../teacher/TeacherAddContentButton";

class SlideHeader extends React.Component {

    render() {
        let teacherEditButtons = (this.props.slideEditMode === true) ?
            <div className="lessonHeaderButtons">
                <TeacherCancelEditSlideButton />
                <TeacherSaveEditsSlideButton />
                <TeacherAddContentButton />
            </div> :
            <div className="lessonHeaderButtons" >
                <TeacherEditSlideButton/>
            </div>

        return <div className="lessonHeader" >
            <span className="lessonHeaderTitle">{this.props.slideHeader}</span>
            {teacherEditButtons}
        </div>
    }
}

const mapStateToProps = state => ({
    slideEditMode: state.slideEditMode,
});

function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({

        }, dispatch)
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(SlideHeader);