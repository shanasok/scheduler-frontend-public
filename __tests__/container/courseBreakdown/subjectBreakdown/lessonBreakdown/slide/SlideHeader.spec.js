import React from "react";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider as ReduxProvider} from "react-redux";
import {Router} from "react-router-dom";
import { createMemoryHistory } from "history";

import SlideHeader from "../../../../../../src/container/courseBreakdown/subjectBreakdown/lessonBreakdown/slide/SlideHeader";

it('renders correctly', () => {

    const pageContent1 = {
        contentNumber: 1,
        content: {
            contentType: "FACT",
            fact: "Test Fact",
            id: 1,
        }
    };
    const currentPage = {
        id: 1,
        pageNumber: 1,
        pageContent: [pageContent1],
    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage],
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLessonContent: lesson,
        currentPage: currentPage,
    }
    let store = mockStore(mockState)

    const history = createMemoryHistory();
    history.location.state = {
        currentLessonContent: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[currentPage],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}>
            <SlideHeader currentPage={currentPage}/>
        </Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children[0].children[0]).toEqual("Edit Slide");

});

it('renders correctly with no pageContent', () => {

    const currentPage = {
        id: 1,
        pageNumber: 1,

    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage],
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLesson: lesson,
        currentPage: currentPage,
    }
    let store = mockStore(mockState)

    const history = createMemoryHistory();
    history.location.state = {
        lesson: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}>
            <SlideHeader currentPage={currentPage}/>
        </Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children[0].children[0]).toEqual('Edit Slide');
});


it('renders cancel and save buttons when slideEditMode is true', () => {

    const currentPage = {
        id: 1,
        pageNumber: 1,

    };

    const lesson = {
        id: 1,
        name: "test lesson name",
        lessonNumber: 1,
        pages: [currentPage],
    };

    const mockStore = configureMockStore()
    const mockState = {
        currentLesson: lesson,
        currentPage: currentPage,
        slideEditMode: true,
    }
    let store = mockStore(mockState)

    const history = createMemoryHistory();
    history.location.state = {
        lesson: {
            id: 1,
            name: "Threading the Machine",
            lessonNumber: 1,
            pages:[],
        }
    };

    const tree = renderer
        .create(<ReduxProvider store={store}><Router history={history}>
            <SlideHeader currentPage={currentPage}/>
        </Router></ReduxProvider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0].children[0].children[0].children[0]).toEqual('Cancel Edits');
    expect(tree.children[0].children[2].children[0]).toEqual('Save Changes');
});