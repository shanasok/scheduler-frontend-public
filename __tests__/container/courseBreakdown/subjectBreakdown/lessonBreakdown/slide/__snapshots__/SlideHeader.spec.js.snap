// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders cancel and save buttons when slideEditMode is true 1`] = `
<div
  className="lessonTitle"
>
  <div>
    <div>
      <button
        className="cancelEditSlideButton curvedButton"
        onClick={[Function]}
      >
        Cancel Edits
      </button>
    </div>
     
    <button
      className="saveEditsButton curvedButton"
      onClick={[Function]}
    >
      Save Changes
    </button>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="lessonTitle"
>
  <div>
    <button
      className="curvedButton"
      onClick={[Function]}
    >
      Edit Slide
    </button>
  </div>
</div>
`;

exports[`renders correctly with no pageContent 1`] = `
<div
  className="lessonTitle"
>
  <div>
    <button
      className="curvedButton"
      onClick={[Function]}
    >
      Edit Slide
    </button>
  </div>
</div>
`;
